crates/core/src/protocols/evalcheck/subclaims.rs

```diff
@@ -183,9 +207,33 @@ where
 	Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
+pub fn process_composite_sumcheck<U, F>(
+	comp: &CompositeMLE<F>,
+	meta: ProjectedBivariateMeta,
+	eval: F,
+	witness_index: &mut MultilinearExtensionIndex<U, F>,
+	constraint_builders: &mut Vec<ConstraintSetBuilder<F>>,
+	eq_mle: MultilinearExtension<PackedType<U, F>>,
+) -> Result<(), Error>
+where
+	U: UnderlierType + PackScalar<F>,
+	F: TowerField,
+{
+	if !witness_index.has(meta.multiplier_id) {
+		witness_index.update_multilin_poly(vec![(
+			meta.multiplier_id,
+			MLEDirectAdapter::from(eq_mle).upcast_arc_dyn(),
+		)])?;
+	}
+
+	add_composite_sumcheck_to_constraints(meta, constraint_builders, comp, eval);
+	Ok(())
+}
+
 #[derive(Clone, Copy)]
 pub struct ProjectedBivariateMeta {
-	inner_id: OracleId,
+	inner_id: Option<OracleId>, // Some if shifted / packed, None if composite
```

Comment:
if you don't mind, could you please add this comment as a Rust doc comment?
Thank you!